// eslint-config-brad/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const eslint_base_1 = require("./plugins/eslint-base");
const eslint_comments_1 = require("./plugins/eslint-comments");
const jest_1 = require("./plugins/jest");
const import_1 = require("./plugins/import");
const typescript_1 = require("./plugins/typescript");
module.exports = {
    globals: {},
    env: {
        commonjs: true,
        es6: true,
    },
    overrides: [
        {
            // disable certain rules for testing as testing is messy business
            files: [
                'src/**/*.spec.ts',
                'src/**/*.spec.tsx',
                'src/**/*.test.ts',
                'src/**/*.test.tsx',
                'src/**/__mocks__/**/*',
                'src/**/__mock__/**/*',
                'test/**/*',
            ],
            rules: {
                'import/prefer-default-export': 'off',
                'max-len': 'off',
                strict: 'off',
                'typescript/no-explicit-any': 'off',
                'typescript/no-non-null-assertion': 'off',
            },
        },
        {
            // disable certain rules for typings as eslint misdetects them
            files: ['**/*.d.ts'],
            rules: {
                'no-unused-vars': 'off',
                strict: 'off',
                'typescript/no-explicit-any': 'off',
            },
        },
        {
            // allow local tooling scripts to use node modules
            files: [
                'config/**/*.ts',
                'config/**/*.tsx',
                'scripts/**/*.ts',
                'scripts/**/*.tsx',
                'tools/**/*.ts',
                'tools/**/*.tsx',
            ],
            rules: {
                strict: 'off',
                'import/no-nodejs-modules': 'off',
            },
        },
    ],
    plugins: [eslint_comments_1.default.name, import_1.default.name, jest_1.default.name, typescript_1.default.name],
    parser: '@typescript-eslint/parser',
    parserOptions: {
        ecmaVersion: 2018,
        ecmaFeatures: {
            globalReturn: false,
        },
        sourceType: 'module',
    },
    rules: Object.assign({}, eslint_base_1.default.rules, eslint_comments_1.default.rules, jest_1.default.rules, import_1.default.rules, typescript_1.default.rules),
    settings: Object.assign({}, import_1.default.settings),
};
